### PR TITLE
implement TraitInstance to inherit directly from TraitHandler

### PR DESCRIPTION
**Pull request by shoeb-github**
contributes to #697

- copied the methods from`ThisClass` to `TraitInstance`.

### CODE REVIEW

File: traits/trait_handlers.py [hunk 1]
```diff
@@ -419,6 +428,9 @@ def info(self):
 
         return result
 
+    def info_none(self):
```
Comment (mdickinson) [inline traits/trait_handlers.py]: I think this method can be removed: `info_none` is only used in `ThisClass.__init__`.

File: traits/trait_handlers.py [hunk 2]
```diff
@@ -407,6 +407,15 @@ def validate(self, object, name, value):
 
         self.validate_failed(object, name, value)
 
+    def validate_failed(self, object, name, value):
+        self.error(object, name, value)
+
+    def validate_none(self, object, name, value):
```
Comment (mdickinson) [inline traits/trait_handlers.py]: This method isn't needed.

**Comment by shoeb-github** [comment]
Apart from `validate_none` and `info_none` methods, I also removed `validate_failed` method.

**Comment by mdickinson** [comment]
> I also removed `validate_failed` method.

Yep, I think that's reasonable. My only worry about that would be subclasses of `TraitInstance` that rely on `validate_failed`, but I can't see any place where `TraitInstance` is subclassed in ETS, and it's unlikely to be used at all outside ETS.

**Pull request merged**